hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/Histogram1D.java

```diff
@@ -50,17 +60,29 @@ public void createHistogram(final IColumn column, final IMembershipSet membershi
         }
     }
 
+    public void addItem(final double value, final Object item) {
+        int index = this.bucketDescription.indexOf(value);
+        if (index >= 0)
+            this.buckets[index].add(value,item);
+        else this.outOfRange++;
+    }
+
+    public int getNumOfBuckets() { return this.bucketDescription.getNumOfBuckets(); }
+
     public long getMissingData() { return this.missingData; }
 
     public long getOutOfRange() { return this.outOfRange; }
 
     /**
-     * @return the index's bucket or null if not been initialized yet
+     * @return the index's bucket or exception if doesn't exist.
      */
     public Bucket1D getBucket(final int index) {
         if ((index < 0) || (index >= this.bucketDescription.getNumOfBuckets()))
             throw new IllegalArgumentException("bucket index out of range");
-        return this.buckets[index];
+        Bucket1D res = this.buckets[index];
+        if (res == null)
```

Comment:
Can you add a @NonNull annotation to the return value?
I don't know if we need this null check either.